pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -100,11 +98,29 @@ func (m *spirePlugin) MintX509CA(ctx context.Context, request *upstreamauthority
 }
 
 func (m *spirePlugin) PublishJWTKey(ctx context.Context, req *upstreamauthority.PublishJWTKeyRequest) (*upstreamauthority.PublishJWTKeyResponse, error) {
-	return nil, makeError(codes.Unimplemented, "publishing upstream is unsupported")
-}
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+
+	wCert, wKey, wBundle, err := m.getWorkloadSVID(ctx, m.config)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := m.newNodeClientConn(ctx, wCert, wKey, wBundle)
```

Comment:
may we add timeout to this context?

Reply:
A timeout here would be ineffectual. This method is just doing a non-blocking dial to the workload API and returns immediately.

Reply:
I think the context here ends up in grpc.DialContext for reaching upstream node api? Seems like it would be useful to set a timeout in that case

That being said, it also feels like any timeout or deadline is best set by the caller of `PublishJWTKey` i.e. core?

Reply:
Oh, yes. This is for the node API connection. My wires were crossed. In any case, the context won't do what you think it does because the dialing is non blocking and the context only applies to the connection setup:

> DialContext creates a client connection to the given target. By default, it's a non-blocking dial (the function won't wait for connections to be established, and connecting happens in the background). To make it a blocking dial, use WithBlock() dial option.

> In the non-blocking case, the ctx does not act against the connection. It only controls the setup steps.

Reply:
> That being said, it also feels like any timeout or deadline is best set by the caller of PublishJWTKey i.e. core?

I agree.